### ISSUE TITLE
Add support for dumbbell/split interstitials

### ISSUE THREAD

**Issue by seatonullberg** [SMTG-Bham/doped#56]
Are there any plans to support split interstitial generation? Maybe this is already supported with some advanced configuration but it was not obvious to me from the defect generation tutorial.

**Comment by kavanase** [comment]
Direct initialisation of split interstitial structures isn't automated in `doped`, as there's a lot of configurational complexity and combinatorial explosion of possible structures for this. 
However, we have found that the recommended approach of Voronoi tessellation for initial interstitial sites + structure-searching (with `ShakeNBreak` or otherwise) does quite a good job of this, and has identified these structures for all known cases (some examples are [Te interstitials in CdTe](https://pubs.rsc.org/en/content/articlelanding/2022/fd/d2fd00043a), [Li-Li and O-O (peroxide) dumbbells in LNMO](https://pubs.rsc.org/en/content/articlelanding/2023/ta/d3ta00532a) and split Sb interstitials in [Sb₂O₅](https://pubs.acs.org/doi/10.1021/acs.chemmater.3c03257) – similar to the split Ga interstitials in [Ga₂O₃](https://journals.aps.org/prb/abstract/10.1103/PhysRevB.107.024109).

That said, with the functions provided and defect structures output by `doped`, you could directly initialise some split interstitial configurations if you had a specific motif in mind (e.g. non-dumbbell like; 2 nearby-but-separated interstitials + vacancy -> split interstitial)

**Comment by seatonullberg** [comment]
That sounds like a reasonable approach. Could you direct me to the python tool you mentioned that would identify the largest neighboring void? Identifying the most likely orientation like that was a big part of the challenge when I was considering it. I'll familiarize myself with the codebase some more then open a PR.

**Comment by kavanase** [comment]
Yeah so [Voronoi tessellation](https://en.wikipedia.org/wiki/Voronoi_diagram) is the main approach for this I've seen. We use code for this in `ShakeNBreak` `input.py`: https://github.com/SMTG-Bham/ShakeNBreak/blob/main/shakenbreak/input.py for getting the vertices of the Voronoi polyhedra, that we then place interstitials at. With that code from `scipy.spatial`, you can also get the corresponding Voronoi polyhedra and use this. See also the `VoronoiPolyhedron` class from `pymatgen-analysis-defects` for this: https://github.com/materialsproject/pymatgen-analysis-defects/blob/074cfcdc2d1b17d115cb604d52d1ee3f4d73e7a3/pymatgen/analysis/defects/utils.py#L674

I guess the approach would be to take the voronoi nodes (used as candidate interstitial positions) and corresponding polyhedra, choose the polyhedron with the largest volume whose node is within a cutoff (~3 Å?) of the lattice site and use that as the direction of largest interstitial void?

I feel like I've seen other tools out there for this, which might be better suited to the job, but can't remember these off the top of my head.

**Comment by kavanase** [comment]
@seatonullberg just fyi, this [preprint](https://arxiv.org/abs/2412.19330) may be somewhat related to your thoughts here, using `doped` to generate a different form of stoichiometry-conserving defect complex (split vacancies rather than split interstitials in that case)
